packages/synapse-react-client/src/components/CreateTableViewWizard/CreateTableViewWizard.tsx

```diff
@@ -0,0 +1,382 @@
+      },
+    })
+
+  const { mutateAsync: createColumnModels, error: createColumnModelsError } =
+    useCreateColumnModels()
+
+  const onTableTypeSelected = useCallback(
+    (entityType: EntityType, viewTypeMask?: number) => {
+      setEntityType(entityType)
+      if (viewTypeMask != null) {
+        setViewTypeMask(viewTypeMask)
+      }
+      setStep(getStepAfterTypeSelection(entityType))
+    },
+    [],
+  )
+
+  const onViewSelected = useCallback(() => {
+    setStep('CHOOSE_VIEW_TYPE')
+  }, [])
+
+  const onFinish = useCallback(async () => {
+    if (!entityType) {
+      console.error('entityType was not defined!')
+      setStep('CHOOSE_TABLE_TYPE')
+      return
+    }
+    // Create the column models
+    let createdColumnModels: ColumnModel[] = []
+    if (
+      columnModels &&
+      // Do not create columns if this is a SQL-defined view
+      entityType !== EntityType.MATERIALIZED_VIEW &&
+      entityType !== EntityType.VIRTUAL_TABLE
+    ) {
+      try {
+        createdColumnModels = await createColumnModels(columnModels)
+      } catch (e) {
+        // Error will be handled by the hook, exit early if we encountered one
+        return
+      }
+    }
+    // Create the entity
+    const entityToCreate: Entity = {
+      name,
+      description,
+      parentId: parentId,
+      concreteType: convertToConcreteEntityType(entityType),
+    }
+
+    maybeSetColumnIds(entityToCreate, entityType, createdColumnModels)
+    maybeSetScopeIds(
+      entityToCreate,
+      entityType,
+      entityType === EntityType.SUBMISSION_VIEW
+        ? submissionViewScopeIds
+        : entityViewScopeIds,
+    )
+    maybeSetViewTypeMask(entityToCreate, entityType, viewTypeMask)
+    maybeSetDefiningSQL(entityToCreate, entityType, sql)
+
+    try {
+      await createEntity(entityToCreate)
+    } catch (e) {
+      // Error will be handled by the hook
+      return
+    }
+  }, [
+    columnModels,
+    createColumnModels,
+    createEntity,
+    description,
+    entityType,
+    name,
+    parentId,
+    entityViewScopeIds,
+    submissionViewScopeIds,
+    sql,
+    viewTypeMask,
+  ])
+  const onColumnSchemaSubmit = useCallback(
+    (columnModels: SetOptional<ColumnModel, 'id'>[]) => {
+      setColumnModels(columnModels)
+      setStep('TABLE_NAME')
+    },
+    [],
+  )
+
+  const onColumnSchemaNextClicked = useCallback(() => {
+    if (columnSchemaFormRef.current != null) {
+      columnSchemaFormRef.current.submit()
+    }
+  }, [])
+
+  const showBackButton = step !== 'CHOOSE_TABLE_TYPE'
+  const showNextButton =
+    step !== 'CHOOSE_TABLE_TYPE' && step !== 'CHOOSE_VIEW_TYPE'
+
+  const onBackButtonClicked = useCallback(() => {
+    setStep(getPreviousStep(step, entityType))
+  }, [step, entityType])
+
+  const onNextButtonClicked = useCallback(() => {
+    if (
+      step === 'TABLE_NAME' ||
+      // TODO: After PLFM-8209 is complete, TABLE_SQL can come before TABLE_NAME
+      step === 'TABLE_SQL'
+    ) {
```

Comment:
Could you use the `isLastStep` method here?

Reply:
definitely should!